File: livetranslate-platform/services/asr-service/asr-server/src/main/java/com/livetranslate/asr/application/service/ASRApplicationService.java
package com.livetranslate.asr.application.service;

import com.livetranslate.asr.domain.model.ASRSession;
import com.livetranslate.asr.domain.model.RecognitionResult;
import com.livetranslate.asr.domain.service.QwenASRDomainService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.util.function.Consumer;

/**
 * ASR 应用服务
 * 编排：接收音频 → 调引擎 → 发事件
 */
@Slf4j
@Service
@RequiredArgsConstructor
public class ASRApplicationService {

    private final QwenASRDomainService qwenASRDomainService;

    /**
     * 创建 ASR 会话
     */
    public ASRSession createSession(String apiKey, String targetLang, String model) {
        return qwenASRDomainService.createSession(apiKey, targetLang, model);
    }

    /**
     * 处理音频数据
     */
    public void processAudio(ASRSession session, byte[] pcmData, Consumer<RecognitionResult> callback) {
        qwenASRDomainService.sendAudio(session, pcmData, callback);
    }

    /**
     * 关闭会话
     */
    public void closeSession(ASRSession session) {
        qwenASRDomainService.closeSession(session);
    }
}

File: services/asr-service/asr-server/src/application/service/ASRApplicationService.ts
import { QwenASRDomainService } from '../../domain/service/QwenASRDomainService.js';
import { DashScopeWSClient } from '../../infrastructure/external/DashScopeWSClient.js';
import type { ASRSession } from '../../domain/model/ASRSession.js';

/**
 * ASR 应用服务
 */
export class ASRApplicationService {
  private domainService = new QwenASRDomainService();
  private sessions = new Map<string, { session: ASRSession; client: DashScopeWSClient }>();

  async createSession(apiKey: string, targetLang: string, model: string): Promise<string> {
    const session = this.domainService.createSession(apiKey, targetLang, model);
    const client = new DashScopeWSClient();

    await client.connect(apiKey, model);
    this.sessions.set(session.sessionId, { session, client });

    return session.sessionId;
  }

  sendAudio(sessionId: string, pcmBase64: string): void {
    const entry = this.sessions.get(sessionId);
    if (entry) {
      entry.client.sendAudio(pcmBase64);
    }
  }

  closeSession(sessionId: string): void {
    const entry = this.sessions.get(sessionId);
    if (entry) {
      entry.client.close();
      this.sessions.delete(sessionId);
    }
  }
}
